modules/va_facilities/lib/va_facilities/param_validators.rb

```diff
@@ -34,8 +41,102 @@ def validate_type_and_services_known
       raise Common::Exceptions::InvalidFieldValue.new('services', unknown) unless unknown.empty?
     end
 
+    def validate_street_address
+      if params[:street_address]
+        raise Common::Exceptions::InvalidFieldValue.new('street_address', params[:street_address]) unless
+            params[:street_address].match?(/\d/)
+      end
+    end
+
+    def validate_drive_time
+      Integer(params[:drive_time]) if params[:drive_time]
+    rescue ArgumentError
+      raise Common::Exceptions::InvalidFieldValue.new('drive_time', params[:drive_time])
+    end
+
+    def validate_lat
+      Float(params[:lat]) if params[:lat]
+    rescue ArgumentError
+      raise Common::Exceptions::InvalidFieldValue.new('lat', params[:lat])
+    end
+
+    def validate_lng
+      Float(params[:lng]) if params[:lng]
+    rescue ArgumentError
+      raise Common::Exceptions::InvalidFieldValue.new('lng', params[:lng])
+    end
+
+    def validate_bbox
+      raise ArgumentError unless params[:bbox].nil? || params[:bbox]&.length == 4
+      params[:bbox]&.each { |x| Float(x) }
+    rescue ArgumentError
+      raise Common::Exceptions::InvalidFieldValue.new('bbox', params[:bbox])
+    end
+
+    def validate_a_param_exists(require_one_param)
+      lat_and_long = params.key?(:lat) && params.key?(:long)
+
+      if !lat_and_long && require_one_param.none? { |param| params.key? param }
+        require_one_param.each do |param|
+          unless params.key? param
+            raise Common::Exceptions::ParameterMissing.new(param.to_s, detail: MISSING_FACILITIES_PARAMS_ERR)
+          end
+        end
+      end
+    end
+
+    def valid_location_query?
+      case location_keys
+      when [] then true
+      when %i[lat long] then true
+      when [:state]     then true
+      when [:zip]       then true
+      when [:bbox]      then true
+      else
+        # There can only be one
+        render json: {
+          errors: ['You may only use ONE of these distance query parameter sets: lat/long, zip, state, or bbox']
+        },
+               status: 422
+      end
+    end
+
+    def validate_required_nearby_params(required_params)
+      param_keys = params.keys.map(&:to_sym)
+      address_params = required_params[:address]
+      lat_lng_params = required_params[:lat_lng]
+      address_difference = (address_params - param_keys)
+      lat_lng_difference = (lat_lng_params - param_keys)
+
+      unless valid_params?(address_difference, lat_lng_difference)
+        if ambiguous?(address_difference, lat_lng_difference, address_params, lat_lng_params)
+          raise Common::Exceptions::ParameterMissing.new(detail: AMBIGUOUS_PARAMS_ERR)
```

Comment:
This line can never be reached.
`valid_params?` method eliminates any `address_difference` and `lat_lng_difference` combination from getting through that would allow `ambiguous?` to return `true`.

(There is no spec testing this path. That is, there is no spec testing the case when `AMBIGUOUS_PARAMS_ERR` would be raised.)

Reply:
Here's the scenario where it fails for being ambiguous

https://github.com/department-of-veterans-affairs/vets-api/blob/add_lat_and_lng_wh/modules/va_facilities/spec/validators/param_validators_spec.rb#L462